### PR TITLE
[TBCM-20] Select Specific Care Location Profile

### PR DESCRIPTION
**Pull request by leon-fw**
- Introduced a generic dropdown
- Introduced a temporary component (CareLocationSelector) that makes the GET call to the API and passes the data to the dropdown (probably could be moved to a form later?)
- API call retrieves actual data from the database (the Units table, make sure you have some added if you want to test it).

### CODE REVIEW

File: apps/web/src/components/CareLocationSelector.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/CareLocationSelector.tsx:11]: Could you move line 11 - 19 to a separate hook? And update `useState` and `useEffect` instead of React.

File: apps/web/src/components/CareLocationSelector.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/CareLocationSelector.tsx:23]: Also, only after selecting the radio btn, this drop down will appear. Kindly go through this design https://www.figma.com/file/QuGyUBu1JEkNaGTb8ZfC3l/Care-mapping-design?node-id=173%3A10180

File: apps/web/src/services/carelocations.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/services/carelocations.tsx:4]: Ahhh, I'm a little skeptical about this approach. I'll make a change to this PR, let me know if that looks good.

File: apps/web/src/pages/index.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/pages/index.tsx:8]: We should wait for @PeterPavlatos's PR to be merged before merging this

File: apps/web/src/components/CareLocationSelector.tsx
Comment (leon-fw) [reply apps/web/src/components/CareLocationSelector.tsx:11]: Hey Karthikk, thanks for the feedback!

Sorry, could you please clarify what you mean by "move this to a separate hook", I am kind of new to this? As in, move into a separate file and then import this whole functionality or? Thanks in advance!

File: apps/web/src/components/CareLocationSelector.tsx
Comment (leon-fw) [reply apps/web/src/components/CareLocationSelector.tsx:23]: Right, good point! Are the radio buttons part of this Jira ticket or should it be a separate one?

File: apps/web/src/services/carelocations.tsx
Comment (leon-fw) [reply apps/web/src/services/carelocations.tsx:4]: Please do let me know, thank you!

File: apps/web/src/pages/index.tsx
Comment (leon-fw) [reply apps/web/src/pages/index.tsx:8]: Sure thing, would make sense.

File: apps/web/src/components/CareLocationSelector.tsx
Comment (karthikktamilmani-freshworks) [reply apps/web/src/components/CareLocationSelector.tsx:23]: It's a part of this ticket

Comment (leon-fw) [reply apps/web/src/components/CareLocationSelector.tsx:23]: Thank you, I'll go ahead and add it to the Jira ticket description too, seems to be missing as of now 👍

File: apps/web/src/components/CareLocationSelector.tsx
Comment (karthikktamilmani-freshworks) [reply apps/web/src/components/CareLocationSelector.tsx:11]: Move it to a separate file under hooks or services as useCareLocation(), and export the `careLocations` variable alone from the hook and import it in this file

File: apps/web/src/common/constants.ts
Comment (karthikktamilmani-freshworks) [inline apps/web/src/common/constants.ts:48]: Could you move this to styles.ts and name more appropriately. formFormatting is too generic

File: apps/web/src/components/Dropdown.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/Dropdown.tsx:12]: This entire component should be under `components/generic` which gets defaultValue, options, label and all those stuffs as props

File: apps/web/src/pages/_app.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/pages/_app.tsx:7]: These 2 lines can be removed

**Review by karthikktamilmani-freshworks — COMMENTED** [review]
The text on radio btn and stuffs are blue in color. But it should be black. Also the selected design for radio btn is different from the Designs. Kindly refer

File: apps/web/src/common/constants.ts
Comment (leon-fw) [reply apps/web/src/common/constants.ts:48]: Done

File: apps/web/src/components/Dropdown.tsx
Comment (leon-fw) [reply apps/web/src/components/Dropdown.tsx:12]: Moved to a generic component.

File: apps/web/src/pages/_app.tsx
Comment (leon-fw) [reply apps/web/src/pages/_app.tsx:7]: Removed

File: apps/web/src/components/Dropdown.tsx
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/Dropdown.tsx:13]: I don't think this file will be needed. The code can be copied to CareLocationSelector

**Review by karthikktamilmani-freshworks — APPROVED** [review]

**Pull request merged**
